src/recommendpolicy/helperFunctions.go

```diff
@@ -189,3 +193,93 @@ func initDeploymentWatcher() {
 		}
 	}
 }
+
+func mergePolicies(policies []types.KnoxSystemPolicy, ms types.MatchSpec, name, namespace string, labels LabelMap) types.KnoxSystemPolicy {
+
+	var merged types.KnoxSystemPolicy
+
+	merged.APIVersion = "v1"
+	merged.Kind = "KubeArmorPolicy"
+	merged.Metadata = map[string]string{
+		"name":      fmt.Sprintf("%v-%v", types.HardeningMergePolicy, name),
+		"namespace": namespace,
+	}
+	merged.Spec.Selector.MatchLabels = labels
+
+	for _, policy := range policies {
+		for i := range policy.Spec.File.MatchDirectories {
+			dir := policy.Spec.File.MatchDirectories[i]
+			dir.Action = policy.Spec.Action
+			dir.Severity = policy.Spec.Severity
+			dir.Message = policy.Spec.Message
+			dir.Tags = policy.Spec.Tags
+			merged.Spec.File.MatchDirectories = append(merged.Spec.File.MatchDirectories, dir)
+		}
+	}
+
+	for _, policy := range policies {
+		for i := range policy.Spec.File.MatchPaths {
```

Comment:
why are we looping through all the policies 4 times? can we loop through once and add all the sub for loops under it?
```
for _, policy := range policies {
   for i := range policy.Spec.File.MatchDirectories {
   // code for file matchDirectories
   }
   for i := range policy.Spec.File.MatchPaths {
   // code for file matchPaths
   }
   // for loop for process matchDirectories
  // for loop for process matchPaths
}
```

Reply:
For loop has been optimized.